### PR TITLE
First cut of established hook interfaces

### PR DESCRIPTION
**Pull request by schernysh**

### CODE REVIEW

File: src/main/java/org/prebid/server/hooks/v1/auction/AuctionRequestHook.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package org.prebid.server.hooks.v1.auction;
+
+import org.prebid.server.hooks.v1.Hook;
+
+public interface AuctionRequestHook extends Hook<AuctionRequestPayload, AuctionInvocationContext> {
```
Comment (rpanchyk) [inline src/main/java/org/prebid/server/hooks/v1/auction/AuctionRequestHook.java:5]: Pls reanme to `RawAuctionRequestHook`

**Review by rpanchyk — CHANGES_REQUESTED** [review]

**Review by rpanchyk — APPROVED** [review]

**Comment by muuki88** [comment]
What's the usage for this? I'm currently looking for something that let's us

- add bid floors dynamically within prebid server for all adapters
- add static targetings for AMP auctions ( `hb_source: "s2s"`  in our case )

Is this something that hook infrastructure is capable of?

**Comment by schernysh** [comment]
Hey @muuki88,

This PR is the beginning of the journey to modularity: https://github.com/prebid/prebid-server/issues/1734. It's on an early stage so far so feel free to join this effort - you may want to start with reviewing PRD, feel free to ask questions and suggest also.

Addressing your questions - modularity is envisioned as a response to the needs similar to those you mentioned.

File: src/main/java/org/prebid/server/hooks/v1/auction/AuctionRequestPayload.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package org.prebid.server.hooks.v1.auction;
+
+import com.iab.openrtb.request.BidRequest;
+
+public interface AuctionRequestPayload {
+
+    BidRequest bidRequest();
```
Comment (muuki88) [inline src/main/java/org/prebid/server/hooks/v1/auction/AuctionRequestPayload.java:7]: Is the BidRequest mutable and can be modified the hook?

**Comment by muuki88** [comment]
Thanks a lot for the pointers. I'll take a look at the documents and see where I can give valuable feedback.

Regarding this PR. As an external contributor I would love to see more documentation on the interfaces and how they are supposed to work, what I need to do as an implementor and where I can take a look at the whole picture (e.g. the google doc).

**Comment by bretg** [comment]
@muuki88 - thought I'd answered this, but maybe not

> add static targetings for AMP auctions ( hb_source: "s2s" in our case )

static AMP targets are supported -- https://docs.prebid.org/prebid-server/endpoints/openrtb2/pbs-endpoint-auction.html#custom-targeting-pbs-java-only

**Pull request merged**